### DISCUSSION TITLE
Is it possible to add SCRAM username and password authentication in Kafka for Stellio Context Broker?

### DISCUSSION THREAD

**Discussion by Gianniskourentzis** [stellio-hub/stellio-context-broker#1196] [Q&A] [answered]
Hello,

I’m using the Stellio Context Broker for my project and I’m looking to enhance the security of our Kafka setup. I would like to know if it is possible to add SCRAM (Salted Challenge Response Authentication Mechanism) username and password authentication in Kafka for the Stellio Context Broker.

Specifically, I am looking for guidance on:

Whether the current version of Stellio Context Broker supports SCRAM authentication for Kafka.
1. Any configuration steps or documentation available for implementing this authentication method.
2. Potential challenges or considerations when setting up SCRAM authentication in this context. 
 

Thank you!

Best regards,
Giannis

**Comment by bobeal** [comment, marked as answer]
Hi @Gianniskourentzis,

Behind the scenes, Stellio uses [Spring for Apache Kafka](https://docs.spring.io/spring-kafka/reference/index.html) to handle and manage the communication with Kafka. So anything that is possible with this lib will be possible in Stellio :)

It should only be a matter of adding the expected configuration properties in https://github.com/stellio-hub/stellio-context-broker/blob/develop/shared/src/main/resources/shared.properties. However, I just searched a bit and I can't find a good documentation on how to correctly setup this type of authentication...

Actually, I just found this gist: https://gist.github.com/lbroudoux/86cf97a93dfdddfdda9f4699ba81728d (and similar snippets). You can also find all existing configuration properties in https://docs.spring.io/spring-boot/appendix/application-properties/index.html.

So it should be possible, but it may need a bit of tries.

Hope this helps.

Benoit.

**Reply by Gianniskourentzis** [reply]
Hi Benoit,

Thank you for the detailed explanation and the resources. This is very helpful! I'll go through the links and give it a try.

Best regards,
Giannis
